docs/_docs/services/services.adoc

```diff
@@ -291,6 +294,78 @@ instead (`IgniteServices.serviceProxy(...)`).
 an issue for real jobs like working with a DB or a cache.
 ====
 
+== Service Middleware
+
+WARNING: This feature is experimental and may change in future releases.
+
+[CAUTION]
+====
+This feature may affect performance of service execution.
+====
+
+Apache Ignite allows the user to implement custom middleware logic for his services and provides the following features:
```

Comment:
Apache Ignite allows the user to implement custom middleware logic for his services and provides the following features --> Apache Ignite allows users to implement custom middleware logic for their services and provides the following features

Reply:
Done, thanks